cmd/common/image/Makefile.build

```diff
@@ -13,9 +13,26 @@ TARGETS = \
 	#
 
 .PHONY: all
-all: $(TARGETS)
+all: $(TARGETS) wasm
 
 $(OUTPUTDIR)/%.bpf.o: $(EBPFSOURCE)
 	$(CLANG) $(BASECFLAGS) $(CFLAGS) -D __TARGET_ARCH_$(subst amd64,x86,$*) \
 		-c $< -I /usr/include/gadget/$*/ -o $@
 	$(LLVM-STRIP) -g $@
+
+.PHONY: wasm
+ifeq ($(WASM),)
+wasm:
+	@echo "No wasm file found. Nothing to do."
+else ifeq (go,$(patsubst %.go,go,$(WASM)))
+wasm: $(WASM)
+	tinygo build -o $(OUTPUTDIR)/program.wasm -target=wasi --no-debug $^
+else ifeq (wasm,$(patsubst %.wasm,wasm,$(WASM)))
+wasm: $(WASM)
+	cp $^ $(OUTPUTDIR)/program.wasm
+else
+wasm:
+	@echo "Supported wasm file types: .go, .wasm"
+	@echo "Unsupported file type: $(notdir $(WASM))."
+	@exit 1
+endif
```

Comment:
Ok, I will try that way and add the environment variable `$(WASM)` in the `all` target.

Reply:
Actually your suggestion does not work because the target name (left of `:`) should be `$(OUTPUTDIR)/program.wasm` for both. And the %.go file is the dependency (right of `:`).

```
$(OUTPUTDIR)/program.wasm: program.go
	tinygo build -o $(OUTPUTDIR)/program.wasm -target=wasi --no-debug $@

$(OUTPUTDIR)/program.wasm: program.wasm
	cp $@ $(OUTPUTDIR)/program.wasm
```

And Makefile is unable to select the correct target depending on the available file (`warning: overriding recipe for target`, `warning: ignoring old recipe for target`).

Alternatively, I could write the if/else/then checks in shell inside an unique target, but I don't see the benefit. It would make the output more complicated to understand in case of error.

Reply:
Ok, I removed the echo. I need to keep the rule to distinguish with the error case when there is an invalid file type.

Reply:
I'm wondering if we can use a different approach by moving more logic to build.go:

1. Define a explicit `wasm` target, not part of `all`
 ```
.PHONY: wasm
wasm: $(WASM)
	tinygo build -o $(OUTPUTDIR)/program.wasm -target=wasi --no-debug $^
```

2. Add logic to cmd/common/image/build.go to only invoke this if the wasm file is golang and needs to be compiled, otherwise it can just be copied by build.go. (It seems a bit weird to me to use the makefile just to copy a file)

It could be also possible to have wasm part of `all`, by using this https://stackoverflow.com/a/23964298, but I'm not totally sure if that works fine.

Reply:
> Define a explicit wasm target, not part of all

I think that would generate more lines of code: build.go has two code paths to call the Makefile: `buildLocal` and `buildInContainer`. So we would need to decide the target (`all` or `all wasm`) in both cases.

> Add logic to cmd/common/image/build.go to only invoke this if the wasm file is golang and needs to be compiled, otherwise it can just be copied by build.go. (It seems a bit weird to me to use the makefile just to copy a file)

I changed the logic to avoid the copy in the Makefile.